mod.go

```diff
@@ -108,22 +115,40 @@ func getVCSInfoForModule(m *listModule) (*moduleVCSInfo, error) {
 	return info, nil
 }
 
-// moduleDir returns the path to the directory to be
-// used for storing the module with the given path.
-func moduleDir(module string) string {
-	// TODO decide what color this bikeshed should be.
-	d := filepath.FromSlash(os.Getenv("GOHACK"))
+// moduleDir returns the path to the directory to be used for storing the
+// module with the given path, as well as the filepath to be used in a replace
+// directive. If $GOHACK is set then it will be used. A relative $GOHACK will
+// be interpreted relative to main module directory.
+func moduleDir(module string) (path string, replPath string, err error) {
+	modfp := filepath.FromSlash(module)
+	// Performing a FromSlash here allows /-delimited path values in environment
```

Comment:
This comment is no longer relevant AFAICS.

Reply:
Thanks for the catch.